crates/app-data/src/app_data.rs

```diff
@@ -31,25 +32,27 @@ pub struct ProtocolAppData {
 /// use of flashloans to settle the associated order.
 /// Since using flashloans introduces a bunch of complexities
 /// all these hints are not binding for the solver.
-#[derive(Clone, Debug, Default, Deserialize, Eq, PartialEq)]
+#[derive(Clone, Debug, Default, Serialize, Deserialize, Eq, PartialEq)]
```

Comment:
Piping the same struct from one end to another turned out to be problematic more often than not in the past.
This means we'd strictly tie how the user reports their flashloan hints to how solvers receive them which may not be what is best in terms of the API.
That's why we have a bunch of DTOs.